lib/diyinject.rb

```diff
@@ -0,0 +1,35 @@
+class Array
+
+  def diy_inject(initial=:no_initial_given, symbol="no_symbol_given")
+
+    return self.diy_inject { |x,y| x + y} if initial == :+
+    return self.diy_inject { |x,y| x * y} if initial == :*
+    return self.diy_inject { |x,y| x - y} if initial == :-
+    return self.diy_inject { |x,y| x / y} if initial == :/
+    return self.diy_inject { |x,y| x ** y} if initial == :**
+    return self.diy_inject { |x,y| x % y} if initial == :%
+
```

Comment:
I see what you've done here, but watch out when you start to get massive blocks of code that look like this! You can check if the initial is a symbol by using something like `initial.class` or `initial.is_a?(Symbol)`